### PR TITLE
CGP 11: Increase the block gas limit to 13M

### PR DESCRIPTION
**Pull request by nambrot**
# CGP [0011]: Increase the gasLimit to 13M

- Date: 2020-09-23
- Author(s): @nambrot
- Status: DRAFT
- Governance Proposal ID #: [if submitted]
- Date Executed: [if executed]

## Overview

This proposal sets the block gas limit parameter of the Celo blockchain to 13M from 10M. This change is necessary to accomodate the contract deploy of Celo core smart contracts in the upcoming first upgrade.

At genesis, the block gas limit was actually 20M to accomodate the initial deploy as well. After the deploy of the Celo core smart contracts, the block gas limit was set down to a more conservative 10M. Since then, Javier Corteso (@jcortejoso) and Kevin Jue (@kevjue) have conducted some load-testing to determine whether raising the block gas limit is indeed safe (read more in the Testing section).

A side-effect of this proposal is the increase of transaction throughput by way of the increased block space by 30%.

## Proposed Changes

1. Increase the block gas limit parameter
  - Destination: Call `setBlockGasLimit` on the `BlockchainParameters` smart contract owned by `Governance`
  - Data: 13 000 0000 (13 Million)
  - Value: 0 (NA)

## Testing

TODO: Describe the testing that was done to ensure this is a safe change

## Verification

TODO: Add confirmation commands once proposal is live

## Risks

- While metrics in the testing environment indicated that consensus and block production remain within expected bounds, generally speaking, increases in block space mean higher computational needs which could impact consensus and block production under adversarial conditions that were not tested.
- Increasing the block space could lead to a quicker growth of chain state which in the long run makes running full nodes more resource-intenstive and could affect decentralization adversely.

## Useful Links

* [Block gas limit entry in the genesis block](https://github.com/celo-org/celo-monorepo/blob/master/packages/celotool/genesis_rc1.json#L23)

### CODE REVIEW

File: CGPs/0011.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- Date: 2020-09-23
+- Author(s): @nambrot
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+
+## Overview
+
+This proposal sets the gasLimit parameter of the Celoblockchain to 13M from 10M. This change is necessary to accomodate the contract deploy of Celo core smart contracts whose size has increased since the initial deploy.
```
Comment (asaj) [inline CGPs/0011.md:11]: missingspace

Comment (asaj) [inline CGPs/0011.md:11]: The size actually hasn't meaningfully increased, the BGL was 20M at the time of contract deployment

File: CGPs/0011.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Proposed Changes
+
+1. Increase the gasLimit
```
Comment (asaj) [inline CGPs/0011.md:15]: nit: block gas limit here and everywhere

File: CGPs/0011.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- Author(s): @nambrot
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+
+## Overview
+
+This proposal sets the block gas limit parameter of the Celo blockchain to 13M from 10M. This change is necessary to accomodate the contract deploy of Celo core smart contracts in the upcoming first upgrade.
+
+At genesis, the block gas limit was actually 20M to accomodate the initial deploy as well. After the deploy of the Celo core smart contracts, the block gas limit was set down to a more conservative 10M. Since then, Javier Corteso (@jcortejoso) and Kevin Jue (@kevjue) have conducted some load-testing to determine whether raising the block gas limit is indeed safe (read more in the Testing section).
```
[the remark this answers is not in the document]
Comment (nambrot) [reply CGPs/0011.md:13]: I clarified it a little bit, though I feel the following sentence + section addresses that?

File: CGPs/0011.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+## Proposed Changes
+
+1. Increase the block gas limit parameter
+  - Destination: Call `setBlockGasLimit` on the `BlockchainParameters` smart contract owned by `Governance`
+  - Data: 13 000 0000 (13 Million)
+  - Value: 0 (NA)
+
+## Testing
+
+To check and validate the performance of the networks under high pressure of gas consumption, a celo testnet was deployed with the next parameters:
```
Comment (m-chrzan) [inline CGPs/0011.md:26]: Nit: "with the *following* parameters" is I think more conventional?

Comment (m-chrzan) [reply CGPs/0011.md:26]: Capitalize "Celo" here and anywhere else it might be inconsistent?

**Review by m-chrzan — APPROVED** [review]
Looks good to me

**Pull request merged**